docs/NextJs.md

```diff
@@ -0,0 +1,181 @@
+
+```jsx
+// in pages/admin.tsx
+import type { NextPage } from "next";
+import dynamic from "next/dynamic";
+
+const App = dynamic(() => import("../src/admin/App"), { ssr: false });
+
+const Admin: NextPage = () => {
+  return <App />;
+};
+
+export default Admin;
+```
+
+Now the admin renders at `http://localhost:3000/admin`, and you can use the Next.js routing system to add more pages.
+
+## Adding an API
+
+[Next.js allows to serve an API](https://nextjs.org/docs/api-routes/introduction) from the same server. You *could* use this to build a CRUD API by hand. However, we consider that building a CRUD API on top of a relational database is a solved problem, and that developers shouldn't spend time reimplemeting it. 
+
+For instance, if you store your data in a [PostgreSQL](https://www.postgresql.org/) database, you can use [PostgREST](https://postgrest.org/en/stable/) to expose the data as a REST API with zero configuration. Even better, you can use a Software-as-a-Service like [Supabase](https://supabase.com/) to do that for you. 
+
+In such cases, the Next.js API can only serve as a Proxy to authenticate client queries, and pass them down to Supabase. 
+
+Let's see an example in practice. 
+
+First, create a Supabase REST API and its associated PostgreSQL database directly on the [Supabase website](https://app.supabase.com/) (it's free for tests and low usage). Once the setup is finished, use the Supabase manager to add the following tables:
+
+- `posts` with fields: `id`, `title`, and `body` 
+- `comments` with fields: `id`, `name`, `body`, and `postId` (a foreign key to the `posts.id` field)
+
+You can populate these tables via the Supabse UI if you want. Supabase exposes a REST API at `https://YOUR_INSTANCE.supabase.co/rest/v1`.
+
+Copy the supabase API URL and service role key into Next.js's `.env.local` file:
+
+```sh
+# In `.env.local`
+SUPABASE_URL="https://MY_INSTANCE.supabase.co"
+SUPABASE_SERVICE_ROLE="MY_SERVICE_ROLE_KEY"
+```
+
+**Tip**: This example uses the **service role key** here and not the anonymous role. This allows write operations without dealing with authorization. **You shouldn't do this in production**, but use the [Supabase authorization](https://supabase.com/docs/guides/auth) feature instead.
+
+Create [a "catch all" API route](https://nextjs.org/docs/api-routes/dynamic-api-routes#optional-catch-all-api-routes) in the Next.js app by adding a `pages/api/admin/[[...slug]].ts` file. This API route redirects all calls from the react-admin app to the Supabase CRUD API:
+
+```jsx
+// in pages/api/admin/[[...slug]].ts
+export default async function handler(req, res) {
+  // get the incoming request URL, e.g. 'posts?limit=10&offset=0&order=id.asc'
+  const requestUrl = req.url.substring("/api/admin/".length);
+  // build the CRUD request based on the incoming request
+  const url = `${process.env.SUPABASE_URL}/rest/v1/${requestUrl}`;
+  const options = {
+    method: req.method,
+    headers: {
```

Comment:
Something I realized in my remix example is that you also need to forward all the original request headers and override only those needed. Otherwise, some mutations do not work

Reply:
We'll fix the example when and if needed